### PR TITLE
[agentserver] Spec compliance: error shapes, session headers, isolation, diagnostic logging, startup config logging, Foundry User-Agent

### PR DESCRIPTION
**Pull request by RaviPidaparthi**
## What

Spec compliance improvements for the agentserver family — `azure-ai-agentserver-responses` v1.0.0b2, `azure-ai-agentserver-core` v2.0.0b2, `azure-ai-agentserver-invocations` v1.0.0b2.

### Error shapes
- Error `code` field now uses spec-compliant values: `"invalid_request_error"` for 400/404, `"server_error"` for 500.
- Deleted-resource errors return HTTP 404 (was 400).
- Cancel terminal-state message updated to `"Cannot cancel a response in terminal state."`.
- SSE replay rejection messages use spec-compliant wording.
- Foundry storage errors explicitly caught and mapped to appropriate HTTP status codes.

### B40 malformed ID error shape
- All endpoints reject malformed response IDs with HTTP 400 before touching storage.
- Error shape updated per spec: `code: "invalid_parameters"`, `param: "responseId{<value>}"` (was `code: "invalid_request_error"`, `param: "response_id"`).
- `previous_response_id` body-field validation also uses `code: "invalid_parameters"`.
- New `invalid_parameters_response()` builder in `_validation.py`.

### x-agent-session-id response header (§8)
- `x-agent-session-id` HTTP response header now set on every protocol endpoint response (success and error).
- POST `/responses`: header echoes the per-request B39-resolved session ID (payload → env var → derivation → random).
- GET / DELETE / CANCEL / INPUT_ITEMS: header echoes `FOUNDRY_AGENT_SESSION_ID` from `host.config`.
- New `_session_headers()` helper on `_ResponseEndpointHandler` for consistent header propagation.

### Eager eviction
- Terminal responses (completed, failed, cancelled, incomplete) are immediately evicted from in-memory runtime state after persistence.
- Subsequent GET/DELETE/Cancel operations fall through to the provider (storage) path.
- `store=false` responses are also evicted (nothing to fall back to → 404).
- `try_evict()` and `mark_deleted()` on `_RuntimeState`.

### Chat isolation enforcement
- When a response is created with `x-agent-chat-isolation-key`, all subsequent operations must include the same key.
- Mismatched or missing keys return an indistinguishable 404.

### Stream flag removal
- Removed stream mode flag stamping on persisted `Response` objects — the persisted response no longer carries the `stream` flag.

### Diagnostic logging
- `InboundRequestLoggingMiddleware` — pure-ASGI middleware logging every inbound HTTP request (method, path, status, duration, correlation headers, OTel trace ID). Status >= 400 → WARNING. Query strings excluded. **Moved from responses to core** so all protocol hosts get it automatically.
- Handler-level INFO logs at all 5 endpoints (create, get, delete, cancel, input_items) with response ID, status, output count.
- Orchestrator handler invocation log with handler function name and response ID.
- Isolation key presence (`has_user_isolation_key`, `has_chat_isolation_key`) logged at all endpoint entry points and in `FoundryStorageLoggingPolicy`.

### Startup configuration logging
- **Core**: `AgentServerHost` lifespan emits 3 INFO-level log lines at startup:
  1. **Platform environment**: `is_hosted`, `agent_name`, `agent_version`, `port`, `session_id`, `sse_keepalive_interval`
  2. **Connectivity** (masked): `project_endpoint` (scheme://host only), `otlp_endpoint` (scheme://host only), `appinsights_configured` (boolean — connection string is **never** logged)
  3. **Host options**: `shutdown_timeout`, registered `protocols`
- **Responses**: Logs `storage_provider` type, `default_model`, `default_fetch_history_count`, `shutdown_grace_period` at construction.
- **Invocations**: Logs `openapi_spec_configured` at construction.
- New `_mask_uri()` helper: strips URIs to scheme://host, returns `"(not set)"` for empty, `"(redacted)"` for unparseable input.

### Storage logging & Foundry User-Agent
- `FoundryStorageLoggingPolicy` — Azure Core pipeline policy for Foundry HTTP call logging with masked URLs (everything before `/storage` redacted), `api-version` query param preserved.
- `_ServerVersionUserAgentPolicy` — lightweight `SansIOHTTPPolicy` that lazily evaluates a callback to set the `User-Agent` header on outbound Foundry storage HTTP requests. Uses `self._build_server_version` (same source of truth as `x-platform-server` response header) so both headers are always identical, including segments from core, responses, and `additional_server_version`.

### Pyright fixes
- Fixed 13 pre-existing pyright errors in production source (0 remaining).
- OTel type stubs → `# type: ignore`, Starlette Middleware factory → `# type: ignore[arg-type]`.

### Ruff lint fixes
- Fixed all I001 (import sorting) and E501 (line too long) across all 3 packages.

### Housekeeping
- Removed .NET references from code comments and docstrings.
- Added cspell ignore words for internal abbreviations (`hdrs`, `myproj`, `myhost`).

## Packages

| Package | Version | Key changes |
|---------|---------|-------------|
| `azure-ai-agentserver-responses` | 1.0.0b2 | Error shapes, isolation, session headers, eager eviction, diagnostic logging, startup config logging, storage logging, Foundry User-Agent |
| `azure-ai-agentserver-core` | 2.0.0b2 | `InboundRequestLoggingMiddleware`, startup config logging, `_mask_uri()`, pyright fixes |
| `azure-ai-agentserver-invocations` | 1.0.0b2 | Startup config logging, core dep `>=2.0.0b2` |

## Tests

- **1055 tests pass** (responses: 861, core: 89, invocations: 105), 5 skipped.
- New test files: `test_chat_isolation_enforcement.py`, `test_malformed_id_validation.py`, `test_eager_eviction.py`, `test_inbound_request_logging.py`, `test_foundry_logging_policy.py`, `test_startup_logging.py` (13 tests: 8 for `_mask_uri`, 5 for startup log assertions including secret non-leakage).
- Updated: `test_session_id_resolution.py` (header assertions on all endpoints), `test_malformed_id_validation.py` (B40 shape assertions).

### CODE REVIEW

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_endpoint_handler.py [hunk 1]
```diff
@@ -674,11 +682,15 @@ async def handle_get(self, request: Request) -> Response:
...
                     await self._provider.get_response(response_id, isolation=_isolation)
                     return _invalid_mode(
-                        "stream replay is not available for this response; to enable SSE replay, "
-                        + "create the response with background=true, stream=true, and store=true",
+                        "This response cannot be streamed because it was not created with background=true.",
                         {},
                         param="stream",
                     )
+                except FoundryResourceNotFoundError:
+                    pass  # Response doesn't exist in provider either — fall through to 404
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_endpoint_handler.py:777]: Fixed — added explicit FoundryBadRequestError catch in the SSE replay fallback path.

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_endpoint_handler.py [hunk 2]
```diff
@@ -874,16 +891,21 @@ async def handle_cancel(self, request: Request) -> Response:
...
                     )
                 if stored_status == "incomplete":
                     return _invalid_request(
-                        "Cannot cancel an incomplete response.",
+                        "Cannot cancel a response in terminal state.",
                         {},
                         param="response_id",
                     )
+            except FoundryResourceNotFoundError:
+                pass  # Fall through to 404 below
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_endpoint_handler.py:1086]: Fixed — added matching FoundryBadRequestError handler in the cancel fallback path.

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_logging_policy.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+import logging
+import time
+from typing import TYPE_CHECKING, TypeVar
+
+from azure.core.pipeline import PipelineRequest, PipelineResponse
+from azure.core.pipeline.policies import AsyncHTTPPolicy
+
+if TYPE_CHECKING:
+    pass
+
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_logging_policy.py]: Fixed — removed the empty TYPE_CHECKING block and import.

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_logging_policy.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+logger = logging.getLogger("azure.ai.agentserver")
+
+T = TypeVar("T")
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_logging_policy.py]: Fixed — removed unused TypeVar import and T declaration.

File: sdk/agentserver/azure-ai-agentserver-responses/CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,25 @@
...
+- Error responses for deleted resources now return HTTP 404 (was 400). Affects `GET /responses/{id}`, `GET /responses/{id}/input_items`, and `DELETE /responses/{id}` (second delete) on previously deleted responses.
+- Cancel on incomplete responses now returns message `"Cannot cancel a response in terminal state."` (was `"Cannot cancel an incomplete response."`).
+- SSE replay rejection messages now use spec-compliant wording:
+  - Non-background responses: `"This response cannot be streamed because it was not created with background=true."`
+  - Background non-stream responses: `"This response cannot be streamed because it was not created with stream=true."`
+
+### Features Added
+
+- `FoundryStorageLoggingPolicy` — Azure Core per-retry pipeline policy that logs Foundry storage HTTP calls (method, URI, status code, duration, correlation headers) at the `azure.ai.agentserver` logger. Replaces the built-in `HttpLoggingPolicy` in the Foundry pipeline to provide single-line summaries with duration timing and error-level escalation.
+
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/CHANGELOG.md]: Fixed — changed to 'warning-level escalation' to match the actual implementation (WARNING for 4xx/5xx).

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_endpoint_handler.py [hunk 6]
```diff
@@ -34,6 +34,7 @@
 from ..models.errors import RequestValidationError
 from ..models.runtime import ResponseExecution, ResponseModeFlags, build_cancelled_response, build_failed_response
 from ..store._base import ResponseProviderProtocol, ResponseStreamProviderProtocol
+from ..store._foundry_errors import FoundryApiError, FoundryBadRequestError, FoundryResourceNotFoundError, FoundryStorageError
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_endpoint_handler.py]: Fixed — removed unused FoundryStorageError from the import.

**Review by vangarp — APPROVED** [review]

File: sdk/agentserver/azure-ai-agentserver-core/azure/ai/agentserver/core/_middleware.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        span = _trace.get_current_span()
+        ctx = span.get_span_context()
+        if ctx and ctx.trace_id:
+            return format(ctx.trace_id, "032x")
+    except Exception:  # pylint: disable=broad-exception-caught
+        pass
+    return None
+
+
+class InboundRequestLoggingMiddleware:
```
Comment (ankitbko) [inline sdk/agentserver/azure-ai-agentserver-core/azure/ai/agentserver/core/_middleware.py:64]: Should we move this to core?

Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-core/azure/ai/agentserver/core/_middleware.py:64]: done

File: sdk/agentserver/azure-ai-agentserver-responses/CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,27 @@
...
+- Handler-level diagnostic logging — all five endpoint handlers (`POST /responses`, `GET /responses/{id}`, `DELETE /responses/{id}`, `POST /responses/{id}/cancel`, `GET /responses/{id}/input_items`) now emit INFO-level logs at entry and on success, including response ID, status, and output count where applicable.
+- Orchestrator handler invocation logging — logs the handler function name and response ID at INFO level before each handler invocation.
+- Chat isolation key enforcement — when a response is created with an `x-agent-chat-isolation-key` header, subsequent GET, DELETE, Cancel, and InputItems requests must include the same key. Mismatched or missing keys return an indistinguishable 404 to prevent cross-chat information leakage. Backward-compatible: no enforcement when the response was created without a key.
+- Malformed response ID validation — all endpoints that accept a `response_id` path parameter now reject malformed IDs (wrong prefix, too short) with HTTP 400 (`invalid_request_error`, `param: "response_id"`) before touching storage. The `previous_response_id` field in POST body is also validated.
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/CHANGELOG.md]: Fixed — updated the CHANGELOG entry to match the actual wire shape: `error.type: "invalid_request_error"`, `error.code: "invalid_parameters"`, `param: "responseId{<value>}"`.

File: sdk/agentserver/azure-ai-agentserver-responses/tests/contract/test_inbound_request_logging.py [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+
+    @app.response_handler
+    def _default_handler(request: Any, context: Any, cancellation_signal: Any):
+        async def _events():
+            if False:  # pragma: no cover
+                yield None
+        return _events()
+
+    if handler is not None:
+        app._create_fn = handler  # noqa: SLF001
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/tests/contract/test_inbound_request_logging.py]: Fixed — replaced `app._create_fn = handler` with `app.response_handler(handler)` to use the public API.

File: sdk/agentserver/azure-ai-agentserver-core/azure/ai/agentserver/core/_base.py [hunk 10]
```diff
@@ -25,6 +33,34 @@
...
+
+
+def _mask_uri(uri: str) -> str:
+    """Return only the scheme and host of a URI, hiding path/query/credentials.
+
+    Returns ``"(not set)"`` for empty or whitespace-only values.
+
+    :param uri: The URI to mask.
+    :type uri: str
+    :return: ``"scheme://host"`` or ``"(not set)"``.
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-core/azure/ai/agentserver/core/_base.py]: Fixed in d281110 — docstring now lists all three return values: `"scheme://host[:port]"`, `"(not set)"`, or `"(redacted)"`.

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_logging_policy.py [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+        try:
+            response = await self.next.send(request)
+        except Exception:
+            elapsed_ms = (time.monotonic() - start) * 1000
+            logger.warning(
+                "Foundry storage %s %s failed after %.1fms "
+                "(client-request-id=%s, has_user_isolation_key=%s, has_chat_isolation_key=%s)",
+                method,
+                url,
+                elapsed_ms,
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_logging_policy.py:110]: Fixed in d281110 — `_mask_storage_url` now strips everything before `/storage` (host, scheme, `/api/projects/{name}` prefix) and drops query params. Only the `/storage/...` resource path is logged. Added 6 unit tests covering project-path redaction, query stripping, and edge cases.

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_endpoint_handler.py [hunk 12]
```diff
@@ -647,55 +729,133 @@ async def handle_get(self, request: Request) -> Response:
...
+                except FoundryApiError as exc:
+                    logger.error("Storage API error for GET response_id=%s: %s", response_id, exc, exc_info=True)
+                    return _error_response(exc, _hdrs)
                 except Exception:  # pylint: disable=broad-exception-caught
                     logger.warning("Provider fallback failed for GET response_id=%s", response_id, exc_info=True)
             else:
+                # Validate starting_after cursor early — invalid cursors must
+                # always get param=starting_after regardless of stream availability.
+                parsed_cursor = self._parse_starting_after(request)
+                if isinstance(parsed_cursor, Response):
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_endpoint_handler.py]: Fixed in d281110 — `_parse_starting_after` now accepts an optional `headers` param so the error response includes `x-agent-session-id`. Both callers in `handle_get` now pass `_hdrs`.

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_endpoint_handler.py [hunk 13]
```diff
@@ -706,9 +866,16 @@ async def handle_get(self, request: Request) -> Response:
             return self._build_live_stream_response(record, parsed_cursor)
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_endpoint_handler.py]: Fixed in d281110 — `_build_live_stream_response` and `_try_replay_persisted_stream` now merge session headers into SSE headers so `x-agent-session-id` is present on all live-stream and replay `StreamingResponse` paths. Cursor-parse errors also include the session header.

File: sdk/agentserver/azure-ai-agentserver-responses/tests/contract/test_malformed_id_validation.py [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+        and fail later (at provider lookup), NOT at format validation."""
+        client = _make_client()
+        valid_id = IdGenerator.new_response_id()
+        r = client.post("/responses", json={
+            "model": "m",
+            "input": [{"role": "user", "content": "hi"}],
+            "previous_response_id": valid_id,
+        })
+        # Should NOT be 400 from format validation — likely 200 or a different error
+        assert r.status_code != 400 or "Malformed" not in r.json().get("error", {}).get("message", "")
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/tests/contract/test_malformed_id_validation.py]: Fixed in d281110 — test now explicitly asserts `error.code != "invalid_parameters"` and `"Malformed" not in message` when status is 400, instead of the weak OR-based assertion.

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_runtime_state.py [hunk 15]
```diff
@@ -33,6 +34,8 @@ async def add(self, record: ResponseExecution) -> None:
         async with self._lock:
             self._records[record.response_id] = record
             self._deleted_response_ids.discard(record.response_id)
+            if record.chat_isolation_key is not None:
+                self._chat_isolation_keys[record.response_id] = record.chat_isolation_key
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/hosting/_runtime_state.py]: Moot — the `_chat_isolation_keys` dict was removed entirely in 01a1cb9. `check_chat_isolation` is now a `@staticmethod` that takes `record.chat_isolation_key` directly; no separate bookkeeping to go stale.

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_provider.py [hunk 16]
```diff
@@ -70,6 +99,12 @@ class FoundryStorageProvider:
     :param settings: Storage settings. If omitted,
         :meth:`~FoundryStorageSettings.from_env` is called automatically.
     :type settings: FoundryStorageSettings | None
+    :param get_server_version: Callable returning the server version string
+        to use as the ``User-Agent`` header on outgoing Foundry HTTP requests.
+        Evaluated lazily on each request so that it reflects the final
+        composed ``x-platform-server`` value.  When ``None`` (default),
+        no ``User-Agent`` is set by this provider.
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_provider.py]: Fixed — updated docstring to say `When None (default), this provider uses Azure Core's default User-Agent policy.` which matches the actual behavior.

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_logging_policy.py [hunk 17]
```diff
@@ -0,0 +1,137 @@
...
+    """Azure Core per-retry pipeline policy that logs Foundry storage calls.
+
+    Logs the HTTP method, masked URI (host redacted, path preserved, query
+    stripped), response status code, duration in milliseconds, and correlation
+    headers (``x-ms-client-request-id``, ``x-ms-request-id``) for
+    observability of storage operations.
+    """
+
+    async def send(self, request: PipelineRequest) -> PipelineResponse:  # type: ignore[override]
+        """Send the request and log the operation details.
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_logging_policy.py]: Fixed — removed the misleading generic type args from `AsyncHTTPPolicy`. The class now inherits `AsyncHTTPPolicy` without parameters (with `# type: ignore[type-arg]`), which is correct since `send` already declares `PipelineRequest` / `PipelineResponse` in its signature.

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_provider.py [hunk 18]
```diff
@@ -81,24 +116,30 @@ def __init__(
...
+        get_server_version: Callable[[], str] | None = None,
     ) -> None:
         self._settings = settings or FoundryStorageSettings.from_env()
+
+        ua_policy: policies.UserAgentPolicy | _ServerVersionUserAgentPolicy
+        if get_server_version is not None:
+            ua_policy = _ServerVersionUserAgentPolicy(get_server_version)
+        else:
+            ua_policy = policies.UserAgentPolicy()
+
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_provider.py:130]: Fixed — restored `sdk_moniker="ai-agentserver-responses/{VERSION}"` in the default `UserAgentPolicy` path. Updated docstring to match.

File: sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_logging_policy.py [hunk 19]
```diff
@@ -0,0 +1,137 @@
...
+            elapsed_ms = (time.monotonic() - start) * 1000
+            logger.warning(
+                "Foundry storage %s %s failed after %.1fms "
+                "(client-request-id=%s, has_user_isolation_key=%s, has_chat_isolation_key=%s)",
+                method,
+                url,
+                elapsed_ms,
+                client_request_id,
+                has_user_isolation_key,
+                has_chat_isolation_key,
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/azure/ai/agentserver/responses/store/_foundry_logging_policy.py:113]: Fixed — added `exc_info=True` to the transport-failure warning so the full traceback is included in logs.

File: sdk/agentserver/azure-ai-agentserver-responses/tests/contract/test_malformed_id_validation.py [hunk 20]
```diff
@@ -0,0 +1,135 @@
...
+# ── Body field ``previous_response_id`` validation ─────────
+
+class TestMalformedPreviousResponseId:
+    """``previous_response_id`` in POST body must be valid ``caresp`` format."""
+
+    def test_malformed_previous_response_id_returns_400_with_details(self) -> None:
+        client = _make_client()
+        r = client.post("/responses", json={
+            "model": "m",
+            "input": [{"role": "user", "content": "hi"}],
```
[the remark this answers is not in the document]
Comment (RaviPidaparthi) [reply sdk/agentserver/azure-ai-agentserver-responses/tests/contract/test_malformed_id_validation.py]: Fixed — renamed test to `test_malformed_previous_response_id_returns_400`, removed false `details` assertion, now asserts `error.type`, `error.code`, and `error.param` which matches the actual wire shape. Updated module docstring too.

**Review by ankitbko — APPROVED** [review]
